grpc/server_test.go

```diff
@@ -0,0 +1,293 @@
+				}
+
+				return &grpchealth.HealthCheckResponse{}, nil
+			},
+			code: codes.OK,
+		},
+	} {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			var events []*sentry.Event
+
+			if err := sentry.Init(sentry.ClientOptions{
+				BeforeSend: func(
+					event *sentry.Event,
+					hint *sentry.EventHint,
+				) *sentry.Event {
+					events = append(events, event)
+					return event
+				},
+			}); err != nil {
+				t.Fatalf("sentry.Init error: %s", err)
+			}
+
+			lis, err := net.Listen("tcp", "localhost:0")
+			if err != nil {
+				t.Fatalf("net.Listen error: %s", err)
+			}
+			defer lis.Close()
+
+			opt := grpc.UnaryInterceptor(sentrygrpc.UnaryServerInterceptor(tt.opts))
+			server := grpc.NewServer(opt)
+			defer server.Stop()
+
+			grpchealth.RegisterHealthServer(
+				server,
+				&mockHealthServer{handler: tt.handler},
+			)
+
+			go func() {
+				if err := server.Serve(lis); err != nil {
+					t.Errorf("grpc serve error: %s", err)
+				}
+			}()
+
+			conn, err := grpc.Dial(lis.Addr().String(), grpc.WithInsecure())
+			if err != nil {
+				t.Fatalf("grpc.Dial error: %s", err)
+			}
+			defer conn.Close()
+
+			client := grpchealth.NewHealthClient(conn)
+
+			req := &grpchealth.HealthCheckRequest{}
+			_, err = client.Check(tt.ctx, req)
+			if w, g := tt.code, status.Code(err); w != g {
+				t.Fatalf("status mismatch: want %s, got %s", w, g)
+			}
+
+			if !sentry.Flush(time.Second) {
+				t.Fatal("sentry.Flush timed out")
+			}
+
+			opts := cmp.Options{
+				cmpopts.IgnoreFields(
+					sentry.Event{},
+					"Contexts",
+					"EventID",
+					"Extra",
+					"Platform",
+					"Sdk",
+					"ServerName",
+					"Tags",
+					"Timestamp",
+				),
+				cmpopts.IgnoreFields(
+					sentry.Request{},
+					"Env",
+				),
+				cmpopts.IgnoreFields(
+					sentry.Exception{},
+					"Stacktrace",
+				),
+			}
+
+			if d := cmp.Diff(tt.events, events, opts); d != "" {
+				t.Fatalf("events mismatch (-want +got):\n%s", d)
+			}
+		})
+	}
+}
+
+type mockHealthServer struct {
+	grpchealth.UnimplementedHealthServer
+
+	handler func(
+		context.Context,
+		*grpchealth.HealthCheckRequest,
+	) (*grpchealth.HealthCheckResponse, error)
+}
+
+func (m *mockHealthServer) Check(
+	ctx context.Context,
+	req *grpchealth.HealthCheckRequest,
+) (*grpchealth.HealthCheckResponse, error) {
+	return m.handler(ctx, req)
+}
+
+func TestReportOnCodes(t *testing.T) {
+	for _, tt := range []struct {
+		name  string
+		err   error
+		codes []codes.Code
+		want  bool
+	}{
+		{
+			name:  "returns true on code match",
+			err:   status.Error(codes.Aborted, ""),
+			codes: []codes.Code{codes.Aborted},
+			want:  true,
+		},
+		{
+			name:  "returns false on code mismatch",
+			err:   status.Error(codes.Aborted, ""),
+			codes: []codes.Code{codes.Canceled},
+			want:  false,
+		},
+	} {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			if w, g := tt.want, sentrygrpc.ReportOnCodes(tt.codes...)(tt.err); w != g {
+				t.Fatalf("ReportOnCodes: want %t, got %t", w, g)
```

Comment:
"ReportOnCodes" is already part of the test name, so we don't need to repeat it in the error message:

```suggestion
			if got := sentrygrpc.ReportOnCodes(tt.codes...)(tt.err); got != tt.want {
				t.Fatalf("got %t, want %t", got, tt.want)
```

Reply:
> "ReportOnCodes" is already part of the test name, so we don't need to repeat it in the error message:

Got it.

I personally prefer to use the variable `w` not to repeat `tt.want`. But I'm OK to change if the suggested style is widely used in the sentry-go codebase.